File: road-runner-quickstart-master/TeamCode/src/main/java/org/firstinspires/ftc/teamcode/robot/actions/IntakeOut.java
package org.firstinspires.ftc.teamcode.robot.actions;

import androidx.annotation.NonNull;

import com.acmerobotics.dashboard.telemetry.TelemetryPacket;
import com.acmerobotics.roadrunner.Action;

import org.firstinspires.ftc.teamcode.robot.Intake;

public class IntakeOut implements Action {
    private boolean init = false;
    private Intake i;

    @Override
    public boolean run(@NonNull TelemetryPacket packet) {
        if(!init) {
            Intake.target = 200;
            init = true;
        }

        return i.iM.getCurrentPosition() < 150;
    }

    public IntakeOut(Intake li) {
        i = li;
    }

    public Action intakeOut(Intake li) {
        return new IntakeOut(li);
    }
}

File: road-runner-quickstart-master/TeamCode/src/main/java/org/firstinspires/ftc/teamcode/dairy/actions/IntakeOut.kt
package org.firstinspires.ftc.teamcode.dairy.actions

import com.acmerobotics.dashboard.telemetry.TelemetryPacket
import com.acmerobotics.roadrunner.Action
import org.firstinspires.ftc.teamcode.dairy.subsystems.Intake
import org.firstinspires.ftc.teamcode.dairy.util.Waiter

class IntakeOut() : Action {
    private var init = false
    private lateinit var waiter: Waiter

    override fun run(p: TelemetryPacket): Boolean {
        if (!init) {
            Intake.target = 300.0
            Intake.pidused = true
            init = true
        }

        return Intake.intake!!.currentPosition >= 280
    }

    companion object {
        fun intakeOut(): Action {
            return IntakeOut()
        }}
}